support-workers/src/typescript/test/zuoraDateReplacer.test.ts

```diff
@@ -1,35 +1,37 @@
-import { zuoraDateReplacer } from '../util/zuoraDateReplacer';
+import { replaceDatesWithZuoraFormatDeep } from '../util/zuoraDateReplacer';
 
-describe('zuoraDateReplacer', () => {
+describe('replaceDatesWithZuoraFormatDeep', () => {
 	const realDate = new Date('2025-09-01T12:25:09Z');
+	const isoString = '2025-10-15T08:30:00Z';
 
-	it('replaces Date objects with formatted string', () => {
-		const obj = { date: realDate };
-		const result = JSON.stringify(obj, zuoraDateReplacer);
-		expect(result).toEqual('{"date":"2025-09-01"}');
+	it('formats Date instances at root', () => {
+		const original = { date: realDate };
+		const result = replaceDatesWithZuoraFormatDeep(original);
+		expect(result.date).toBe('2025-09-01');
+		expect(original.date).toBeInstanceOf(Date);
```

Comment:
QQ: What does this check against, don't you already know it contains type Date?

Reply:
It just checks that the input object hasn't changed, I've updated the test name to make that clear